py/picca/io.py

```diff
@@ -148,127 +111,111 @@ def read_absorbers(filename):
-        ra = hdul[1]['TARGET_RA'][:].astype('float64')
-        dec = hdul[1]['TARGET_DEC'][:].astype('float64')
-        plate = np.array(
-            [int('{}{}'.format(t, p)) for t,p in zip(hdul[1]['TILEID'][:],hdul[1]['PETAL_LOC'][:])])
-        mjd = hdul[1]['NIGHT'][:]
-        fiberid = hdul[1]['FIBER'][:]
-        userprint("using DESI mini SV DRQ catalog format")
-    else:
-        thingid = hdul[1]['THING_ID'][:]
-        ra = hdul[1]['RA'][:].astype('float64')
-        dec = hdul[1]['DEC'][:].astype('float64')
-        plate = hdul[1]['PLATE'][:]
-        mjd = hdul[1]['MJD'][:]
-        fiberid = hdul[1]['FIBERID'][:]
-
-    ## Sanity
+    if 'Z' not in catalog.colnames:
+        if 'Z_VI' in catalog.colnames:
+            catalog.rename_column('Z_VI', 'Z')
+            userprint("Z not found (new DRQ >= DRQ14 style), using Z_VI (DRQ <= DRQ12)")
+        else:
+            userprint("ERROR: No valid column for redshift found in ", drq_filename)
+            return None
+
+    ## Sanity checks
     userprint('')
-    w = np.ones(ra.size, dtype=bool)
-    userprint(" start                 : nb object in cat = {}".format(w.sum()))
-    w &= thingid > 0
-    userprint(" and thingid > 0       : nb object in cat = {}".format(w.sum()))
-    w &= ra != dec
-    userprint(" and ra != dec         : nb object in cat = {}".format(w.sum()))
-    w &= ra != 0.
-    userprint(" and ra != 0.          : nb object in cat = {}".format(w.sum()))
-    w &= dec != 0.
-    userprint(" and dec != 0.         : nb object in cat = {}".format(w.sum()))
-    w &= z_qso > 0.
-    userprint(" and z > 0.            : nb object in cat = {}".format(w.sum()))
+    w = np.ones(len(catalog), dtype=bool)
+    userprint(f" start                 : nb object in cat = {np.sum(w)}")
+    w &= catalog[obj_id_name] > 0
+    userprint(f" and {obj_id_name} > 0       : nb object in cat = {np.sum(w)}")
+    w &= catalog['RA'] != catalog['DEC']
+    userprint(f" and ra != dec         : nb object in cat = {np.sum(w)}")
+    w &= catalog['RA'] != 0.
+    userprint(f" and ra != 0.          : nb object in cat = {np.sum(w)}")
+    w &= catalog['DEC'] != 0.
+    userprint(f" and dec != 0.         : nb object in cat = {np.sum(w)}")
 
     ## Redshift range
-    if not z_min is None:
-        w &= z_qso >= z_min
-        userprint((" and z >= z_min        : nb object in cat"
-                   "= {}".format(w.sum())))
-    if not z_max is None:
-        w &= z_qso < z_max
-        userprint((" and z < z_max         : nb object in cat"
-                   "= {}".format(w.sum())))
+    w &= catalog['Z'] >= z_min
+    userprint(f" and z >= {z_min}        : nb object in cat = {np.sum(w)}")
+    w &= catalog['Z'] < z_max
+    userprint(f" and z < {z_max}         : nb object in cat = {np.sum(w)}")
 
     ## BAL visual
-    if not keep_bal and bi_max is None:
-        try:
-            bal_flag = hdul[1]['BAL_FLAG_VI'][:]
+    if keep_bal == False and bi_max is None:
```

Comment:
instead of `keep_bal == False` the standard is to type `not keep_bal`

Reply:
ok